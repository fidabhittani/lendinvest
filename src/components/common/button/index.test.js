import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import CustButton from "./index";

describe("Component: CustButton", () => {
  const onClick = jest.fn();
  const props = {
    text: "invest",
    onClick,
  };
  const btn = shallow(<CustButton {...props} />);

  it("renders correctly", () => {
    const tree = renderer.create(<CustButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a button with invest as text", () => {
    expect(btn.text()).toBe("invest");
  });

  it("button element click handle", () => {
    btn.children().simulate("click");
    expect(onClick).toHaveBeenCalled();
  });
});
