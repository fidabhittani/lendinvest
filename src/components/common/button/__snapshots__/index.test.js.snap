// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CustButton renders correctly 1`] = `
<div
  className="lend-invest-btn"
>
  <button
    onClick={[MockFunction]}
  >
    invest
  </button>
</div>
`;
